evals/scripts/compare_rules.py

```
@@ -0,0 +1,438 @@
+            raise SystemExit(f"{path}:{lineno}: invalid JSON: {exc}") from exc
+        for req in ("id", "category", "query", "reference", "rubric", "checks"):
+            if req not in case:
+                raise SystemExit(f"{path}:{lineno}: case missing required field {req!r}")
+        if case["category"] not in CATEGORIES:
+            raise SystemExit(f"{path}:{lineno}: unknown category {case['category']!r}")
+        # Validate the shape of `checks` so a malformed dataset fails fast here
+        # rather than crashing later inside deterministic_fails (e.g. checks=null,
+        # or a non-string in must_not_contain calling .lower()).
+        checks = case["checks"]
+        if not isinstance(checks, dict):
+            raise SystemExit(f"{path}:{lineno}: 'checks' must be an object, got {type(checks).__name__}")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)